File: src/test/java/net/pterodactylus/rhynodge/filters/comics/BusinessCatComicFilterTest.java
package net.pterodactylus.rhynodge.filters.comics;

import java.io.IOException;

import net.pterodactylus.rhynodge.filters.ResourceLoader;

import com.google.common.base.Optional;
import org.hamcrest.MatcherAssert;
import org.hamcrest.Matchers;
import org.jsoup.nodes.Document;
import org.junit.Test;

/**
 * Unit test for {@link BusinessCatComicFilter}.
 */
public class BusinessCatComicFilterTest {

	private final BusinessCatComicFilter filter = new BusinessCatComicFilter();
	private final Document document;

	public BusinessCatComicFilterTest() throws IOException {
		document = ResourceLoader.loadDocument(getClass(), "business-cat.html", "https://www.businesscatcomic.com/");
	}

	@Test
	public void imageTitleCanBeExtracted() {
		MatcherAssert.assertThat(filter.extractTitle(document), Matchers.is(Optional.of("Full Circle")));
	}

	@Test
	public void imageUrlsCanBeExtracted() {
		MatcherAssert.assertThat(filter.extractImageUrls(document), Matchers.contains("https://www.businesscatcomic.com/wp-content/uploads/2019/11/2018-09-07-Full-Circle.png"));
	}

	@Test
	public void imageCommentsCanBeExtracted() {
		MatcherAssert.assertThat(filter.extractImageComments(document), Matchers.empty());
	}

}

File: src/test/kotlin/net/pterodactylus/rhynodge/filters/comics/BusinessCatComicFilterTest.kt
package net.pterodactylus.rhynodge.filters.comics

import net.pterodactylus.rhynodge.filters.ResourceLoader
import net.pterodactylus.rhynodge.states.ComicState
import net.pterodactylus.rhynodge.states.HtmlState
import org.hamcrest.MatcherAssert.assertThat
import org.hamcrest.Matchers.contains
import org.junit.Test

/**
 * Unit test for [BusinessCatComicFilter].
 */
class BusinessCatComicFilterTest {

	@Test
	fun `comic is extracted correctly`() {
		val comicState = filter.filter(htmlState) as ComicState
		assertThat(comicState.comics(), contains(
				ComicState.Comic("Full Circle").add(ComicState.Strip("https://www.businesscatcomic.com/wp-content/uploads/2019/11/2018-09-07-Full-Circle.png", ""))
		))
	}

	private val filter = BusinessCatComicFilter()
	private val htmlState: HtmlState = ResourceLoader
			.loadDocument(javaClass, "business-cat.html", "https://www.businesscatcomic.com/")
			.let { HtmlState("https://www.businesscatcomic.com/", it) }

}
